### PR TITLE
Fix #1493 Path#add crashes whith 1000000 segments

### LINKED ISSUE #1493: Maximum call stack size exceeded when cloning a path with many segments
# Description/Steps to reproduce

 1. Create a path with more than 125122 segments
 2. Clone it

# Link to reproduction test-case

http://sketch.paperjs.org/#S/VY7BDoIwEER/ZdPEAIEQwXgBOfkDJh48iIdCV9sUuwSqHAj/bku8sKfZmczbnZnhb2QFu2q0rWQJa0n4vUMLPbcSKjA4wcXJMKqNt9LRDqTxTB0NLg6ajrc6KDfhTYm1mzn7SUPocaraQwnqlOXHLD94GccRzLUBN2uXCxGq3VfhlDb0MWJMJ89JYGtKVC9pI4de/kfbjgyGUen+bwbkuidl7MiK+2P5AQ==

# Note

It is possible to get around this problem by splitting the path before cloning it.

### PR DESCRIPTION
**Pull request by sasensi**
### Description

This change prevent `Maximum call stack exceeded` error by using a loop to add segment instead of passing segments array as argument.

#### Related issues

- Fixes #1493

### Checklist

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the JSHint rules (`npm run jslint` passes)

### CODE REVIEW

File: src/path/Path.js [hunk 1]
```diff
@@ -404,8 +404,10 @@ var Path = PathItem.extend(/** @lends Path# */{
                 this._updateSelection(segment, 0, segment._selection);
         }
         if (append) {
-            // Append them all at the end by using push
-            segments.push.apply(segments, segs);
+            // Append them all at the end.
+            // Use a loop as it is the best way to handle big arrays (see #1493)
+            for (var i = 0; i < segs.length; i++)
+                segments.push(segs[i]);
```
Comment (lehni) [inline src/path/Path.js]: Some code-style remarks:

- Let's use blocks after `if` and `for` (unless the statement is super simple). I know the code-base doesn't do this currently in some places, but I tend to disagree with this by now. We shouldn't minify code by hand :) 

- Let's use this loop optimisation: `for (var i = 0, l = segs.length; i < l; i++)`

Result:

```js
for (var i = 0, l = segs.length; i < l; i++) {
    segments.push(segs[i]);
}
```

File: test/tests/Path.js [hunk 2]
```diff
@@ -611,3 +611,13 @@ test('Path#arcTo(from, through, to); where from, through and to all share the sa
     }
     equals(error != null, true, 'We expect this arcTo() command to throw an error');
 });
+
+test('#1493 Path#add with 1000000 segments', function() {
+    var path = new Path();
+    for (var i=0; i<1000000; i++)
+    {
```
Comment (lehni) [inline test/tests/Path.js]: Could you bring the bracket to the same line as the `for`?  That's the convention in the paper.js code-base

File: src/path/Path.js (hunk 1 above)
Comment (sasensi) [reply src/path/Path.js]: - Sorry again for the style, I put that in my IDE rules too :)
- OK for the loop optimisation

File: test/tests/Path.js (hunk 2 above)
Comment (sasensi) [reply test/tests/Path.js]: I added this one too.

**Comment by sasensi** [comment]
Sure, I'd love to be part of the team, I have actually plenty of time to put into the project so I'll do my best to help.
Just, as this will be my first time as an open source project contributor, I might made small mistakes (as in those code style mistakes...) ;)

**Comment by sasensi** [comment]
Ok, I'll do that for future commits.

**Pull request merged**